cpp/src/arrow/python/sequence.h

```diff
@@ -0,0 +1,137 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#ifndef PYTHON_ARROW_SEQUENCE_H
+#define PYTHON_ARROW_SEQUENCE_H
```

Comment:
I implemented the first solution now. Originally I made the headers sequence.h and dict.h so that Objects can be constructed from C++ code that will be valid Python objects and can be deserialized in Python. However we should probably think some more about the API before we expose this.